ext/opcache/jit/zend_jit_internal.h

```diff
@@ -717,8 +717,13 @@ static zend_always_inline const zend_op* zend_jit_trace_get_exit_opline(zend_jit
 	return NULL;
 }
 
-static zend_always_inline bool zend_jit_may_be_polymorphic_call(const zend_op *opline)
+static zend_always_inline bool zend_jit_may_be_polymorphic_call(const zend_op *opline, const zend_function *func)
 {
+	/* function can change unless it is preloaded */
+	if (!func || !(func->common.fn_flags & ZEND_ACC_PRELOADED)) {
+		return 1;
+	}
```

Comment:
We could also consider that `func` can not change if it comes from SHM and `opcache.validate_timestamps` is Off